weather_forecast_retrieval/data/hrrr/http_retrieval.py

```diff
@@ -90,62 +88,110 @@ def fetch_by_date(self, start_date=None, end_date=None):
             out_path = self.output_dir
         self.out_path = out_path
 
-        url_date = HttpRetrieval.URL.format(
-            self.start_date.strftime(FileHandler.SINGLE_DAY_FORMAT)
-        )
+    def fetch_by_date(self, start_date, end_date, forecast_hour=None):
+        """Fetch data from NOMADS between a date range for a given forecast hour.
+        The default will download all forecast hours.
+
+        Args:
+            start_date (str): start date string
+            end_date (str): end date string
+            forecast_hour (list, optional): list of forecast hours. Defaults to None.
+
+        Returns:
+            list: list of responses for data downloads
+        """
+
+        self.log.info('Retrieving data from the http site')
+
+        self.start_date = start_date
+        self.end_date = end_date
+        self.forecast_hour = forecast_hour
+
+        self.check_dates()
```

Comment:
From a design perspective, is it necessary to set the dates and hour as an instance variable?
Seems like you could change the `check_dates` method to pass that in as an argument, return valid dates, and then use those in the method

Reply:
I think it is because it's used so much in creating url's, file names, etc. If it wasn't set they they would need to be passed around to almost every method.

Plus, it was already set on the instance in [the ConfigFile class](https://github.com/USDA-ARS-NWRC/weather_forecast_retrieval/blob/8f34d8259b23998b8483ab069fbd23cd22b81980/weather_forecast_retrieval/data/hrrr/config_file.py#L30).